url284ba87ea1_swabhs_forbidden_paths_tgz-pJ8-tagging_hypergraph_TaggingHypergraphGeneratorJ8/src/tagging_hypergraph/TaggingHypergraphGenerator.java

```diff
@@ -0,0 +1,264 @@
+package tagging_hypergraph;
+
+import hypergraph.HypergraphProto.Hyperedge;
+import hypergraph.HypergraphProto.Hypergraph;
+import hypergraph.HypergraphProto.Vertex;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.TreeMap;
+import java.util.List;
+import java.util.Map;
+
+import learning.InputReader;
+
+/**
+ * Generates a hypergraph for the tagging task. We use global linear models here, each history
+ * containing information about the trigram. 
+ * Each node in the hypergraph is a bigram - the previous and current labels. Each hyperarc is a
+ * normal arc.
+ * @author swabha
+ * TODO(swabha): Test to see if the weights are as expected.
+ * TODO(swabha): Add the suffix features
+ */
+
+public class TaggingHypergraphGenerator {
+	
+	static final List<String> tags = Arrays.asList("I-GENE", "O");
+	static final int numTags = tags.size();
+	
+	// Vertex Maps
+	private static Map<Integer, List<Integer>> outEdgeMap;
+	private static Map<Integer, List<Integer>> inEdgeMap;
+	private static Map<Integer, String> vertexNameMap;
+	
+	// Edge Maps
+	private static Map<Integer, List<Integer>> childMap;
+	private static Map<Integer, Integer> parentMap;
+	private static Map<Integer, Double> edgeWeightMap;
+	
+	
+	static void fillAllMaps(List<String> tokens) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.